src/Voron/Util/NativeList.cs

```diff
@@ -126,14 +126,14 @@ public void Shrink(int newSize)
 
     public void Initialize(ByteStringContext ctx, int count = 1)
     {
-        var capacity = count == 1 ? 1 : Math.Max(1, Bits.PowerOf2(count));
+        var capacity = count == 1 ? 1 : Math.Max(1, Bits.NextAllocationSize(count));
         ctx.Allocate(capacity * sizeof(T), out _storage);
         Capacity = _storage.Length / sizeof(T);
     }
     
     public void Grow(ByteStringContext ctx, int addition)
     {
-        var capacity = Math.Max(1, Bits.PowerOf2(Capacity + addition));
+        var capacity = Math.Max(1, Bits.NextAllocationSize(Capacity + addition));
         ctx.Allocate(capacity * sizeof(T), out var mem);
```

Comment:
We should consider `sizeof(T)` when calculating the next allocation size, as an overflow might occur in the `ctx.Allocate` call.

Reply:
The problem is that we would have to write it as:

```
var capacity = Math.Max(1, Bits.NextAllocationSize((Capacity + addition) *  sizeof(T)) / sizeof(T));
```

And that is just... bleh.
The `Allocate` also ends up calling to `NextAllocationSize`, so that is safe from overflows. 

And in general, we have this in other locations as well. We may need to do a proper scan over all those places and refactor it in a more significant manner.

Reply:
Maybe we should add guardian like:

```csharp
var maxCapacity = int.MaxValue / sizeof(T);
PortableExceptions.ThrowIf(capacity > maxCapacity, "Required xxx len, however the max capacity is {xxx}");
```

It should give us better insight into what the system intended to do if an overflow occurs.

Reply:
Good idea